source/server/config/http/lightstep_http_tracer.cc

```diff
@@ -21,8 +21,8 @@ LightstepHttpTracerFactory::createHttpTracer(const Json::Object& json_config,
                                              Upstream::ClusterManager& cluster_manager) {
 
   std::unique_ptr<lightstep::LightStepTracerOptions> opts(new lightstep::LightStepTracerOptions());
-  opts->access_token = server.api().fileReadToEnd(json_config.getString("access_token_file"));
-  StringUtil::rtrim(opts->access_token);
+  opts->access_token = std::string(
```

Comment:
save rtrim() result in a temp and then use opts->access_token.assign(...date(), ...size()) to avoid the temp.

Reply:
Good call!